radicle-node/src/service.rs

```diff
@@ -777,6 +836,25 @@ where
         // TODO
     }
 
+    fn disconnect_unresponsive_peers(&mut self, now: LocalTime) {
+        let stale = self
+            .sessions
+            .negotiated()
+            .filter(|(_, session)| STALE_CONNECTION_TIMEOUT <= now - session.last_active);
+        for (_, session) in stale {
+            self.reactor
+                .disconnect(session.addr, DisconnectReason::Error(SessionError::Timeout));
+        }
+    }
+
+    // Ensure connection health by pinging connected peers.
+    fn keep_alive(&mut self) {
+        let peer_ips: Vec<_> = self.sessions.negotiated().map(|(ip, _)| *ip).collect();
+        for ip in peer_ips {
+            let _ok = self.ping(&ip);
```

Comment:
```suggestion
            self.ping(&ip).ok();
```

Reply:
Maybe we should only ping peers who have a time since `last_active` that is greater than `PING_INTERVAL`?